packages/riipen-ui/src/components/Input.test.jsx

```diff
@@ -0,0 +1,323 @@
+    const wrapper = mount(<Input />);
+
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  describe("default props", () => {
+    it("sets correct default props", () => {
+      const wrapper = mount(<Input />);
+
+      expect(wrapper.find("Input").props().disabled).toEqual(false);
+      expect(wrapper.find("Input").props().multiline).toEqual(false);
+    });
+  });
+
+  describe("classes prop", () => {
+    it("sets custom class", () => {
+      const classes = ["newClass"];
+
+      const wrapper = mount(<Input classes={classes} />);
+
+      expect(
+        wrapper
+          .find("Input")
+          .childAt(0)
+          .hasClass(classes[0])
+      ).toEqual(true);
+    });
+  });
+
+  describe("disabled prop", () => {
+    it("sets disabled class name with valid custom disabled", () => {
+      const disabled = true;
+
+      const wrapper = mount(<Input disabled={disabled} />);
+
+      expect(
+        wrapper
+          .find("Input")
+          .childAt(0)
+          .childAt(0)
+          .hasClass("disabled")
+      ).toEqual(true);
+    });
+
+    it("sets valid custom disabled", () => {
+      const disabled = true;
+
+      const wrapper = mount(<Input disabled={disabled} />);
+
+      expect(
+        wrapper
+          .find("Input")
+          .childAt(0)
+          .childAt(0)
+          .props().disabled
+      ).toEqual(disabled);
+    });
+  });
+
+  describe("error prop", () => {
+    it("sets valid custom error", () => {
+      const error = <span>Error</span>;
+
+      const wrapper = mount(<Input error={error} />);
+
+      expect(
+        wrapper
+          .find("Input")
+          .childAt(0)
+          .childAt(0)
+          .hasClass("error")
+      ).toEqual(true);
+    });
+
+    it("renders error", () => {
+      const error = <span>Error</span>;
+
+      const wrapper = mount(<Input error={error} />);
+
+      expect(wrapper.find("Typography").contains(error)).toEqual(true);
+    });
+  });
+
+  describe("hint prop", () => {
+    it("sets valid custom hint", () => {
+      const hint = "Test";
+
+      const wrapper = mount(<Input hint={hint} />);
+
+      expect(wrapper.find("InputLabel").props().hint).toEqual(hint);
+    });
+
+    it("renders hint", () => {
+      const hint = "Test";
+
+      const wrapper = mount(<Input hint={hint} />);
+
+      expect(wrapper.text()).toEqual(hint);
+    });
+
+    it("does not render InputLabel element when hint is not given", () => {
+      const wrapper = mount(<Input />);
+
+      expect(
+        wrapper
+          .find("Input")
+          .children()
+          .contains("InputLabel")
+      ).toEqual(false);
+    });
+  });
+
+  describe("label prop", () => {
+    it("sets valid custom label", () => {
+      const label = "Test";
+
+      const wrapper = mount(<Input label={label} />);
+
+      expect(
+        wrapper
+          .find("Input")
+          .childAt(0)
+          .childAt(0)
+          .type()
+      ).toEqual(InputLabel);
```

Comment:
I think the expect statement checks that it "renders InputLabel given label" instead of setting a label.

Reply:
Ya I think we can actually change this test to check if `InputLabel` gets the prop of `children` which is set to `"Test"`.